packages/itwinui-css/src/button/borderless.scss

```diff
@@ -29,11 +29,11 @@
 
   // #region Visual adjustments to help align based on the "visible" edge of the button content.
   &[data-iui-shift='left'] {
-    translate: calc(var(--_iui-button-padding-inline) * -1);
+    margin-inline-start: calc(-1 * var(--_iui-button-padding-inline));
```

Comment:
I changed the implementation to use negative `margin`, so that its "original position" also shifts. With `transform`, it only seems to move visually while occupying a false original position.

The benefit of this can be seen when there are other items next to it. For example, other custom actions in information-panel can shift along with the close button, and the container retains its size (not committed in this PR but can be done in the future).

| with `transform` | with `margin` |
| --- | --- |
| ![image](https://github.com/iTwin/iTwinUI/assets/9084735/b4369b0d-7933-40bd-8e1a-789196110dd8) | ![image](https://github.com/iTwin/iTwinUI/assets/9084735/8ef9efbb-a25c-46b5-8632-e96ece692039) |